src/components/UserSelector.tsx

```diff
@@ -2,39 +1,97 @@
+import { useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import classNames from 'classnames';
+
+type Props = {
+  users: User[];
+  setUserSelected: (user: User) => void;
+  userSelected: User | null;
+  handleUserPosts: (user: User) => void;
+  setShowSideBar: (item: number | null) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  setUserSelected,
+  userSelected,
+  handleUserPosts,
+  setShowSideBar,
+}) => {
+  const [showUsers, setShowUsers] = useState(false);
+  const dropdownMenuRef = useRef<HTMLDivElement | null>(null);
```

Comment:
You initialized your state with props. This could lead to bugs in state updates. Please refer to the links provided in the checklist for more information on this issue.